pymc3/distributions/continuous.py

```diff
@@ -924,6 +925,74 @@ def logcdf(self, value):
         )
 
 
+class ZeroSumNormal(Continuous):
+    def __init__(self, sigma=1, zerosum_dims=None, zerosum_axes=None, **kwargs):
+        shape = kwargs.get("shape", ())
+        dims = kwargs.get("dims", None)
+        if isinstance(shape, int):
+            shape = (shape,)
+
+        if isinstance(dims, str):
+            dims = (dims,)
+
+        self.mu = self.median = self.mode = tt.zeros(shape)
+        self.sigma = tt.as_tensor_variable(sigma)
+
+        if zerosum_dims is None and zerosum_axes is None:
+            if shape:
+                zerosum_axes = (-1,)
+            else:
+                zerosum_axes = ()
+
+        if isinstance(zerosum_axes, int):
+            zerosum_axes = (zerosum_axes,)
+
+        if isinstance(zerosum_dims, str):
+            zerosum_dims = (zerosum_dims,)
+
+        if zerosum_axes is not None and zerosum_dims is not None:
+            raise ValueError("Only one of zerosum_axes and zerosum_dims can be specified.")
+
+        if zerosum_dims is not None:
+            if dims is None:
+                raise ValueError("zerosum_dims can only be used with the dims kwargs.")
+            zerosum_axes = []
+            for dim in zerosum_dims:
+                zerosum_axes.append(dims.index(dim))
+        self.zerosum_axes = [a if a >= 0 else len(shape) + a for a in zerosum_axes]
+        self._degrees_of_freedom = np.prod(
+            [s if axis not in zerosum_axes else s - 1 for axis, s in enumerate(shape)]
+        )
+        self._full_size = np.prod(shape)
+        self._rescaling = np.sqrt(self._full_size / self._degrees_of_freedom)
+
+        # if "transform" not in kwargs or kwargs["transform"]==None:
+        #     kwargs["transform"]=transforms.ZeroSumTransform(zerosum_axes)
+
+        super().__init__(**kwargs)
+
+    def logp(self, value):
+        return Normal.dist(sigma=self.sigma / self._rescaling).logp(value)
```

Comment:
I think this would be cleaner if we could define the logp on the transformed space. I don't think we implemented a way of doing this yet, did we?